ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -586,6 +590,17 @@ public void setRetryDelayFunction(final RetryDelayFunction retryDelayFunction) {
     this.retryDelayFunction = retryDelayFunction;
   }
 
+  public void handleBondingRequest(final DiscoveryPeer peer) {
+    final DiscoveryPeer peerToBond =
+        peerTable.get(peer).filter(known -> known.discoveryEndpointMatches(peer)).orElse(peer);
+
+    if (peerPermissions.allowOutboundBonding(peerToBond)
+        && !PeerDiscoveryStatus.BONDED.equals(peerToBond.getStatus())
+        && !PeerDiscoveryStatus.BONDING.equals(peerToBond.getStatus())) {
```

Comment:
(nit) I think we could simplify:
```suggestion
    if (peerPermissions.allowOutboundBonding(peerToBond)
        && .peerToBond.getStatus().equals(PeerDiscoveryStatus.KNOWN)) {
```